common/class.tx_dlf_solr.php

```diff
@@ -720,30 +741,42 @@ public function __set($var, $value) {
      */
     protected function __construct($core) {
 
-        // Load class.
-        if (!class_exists('Apache_Solr_Service')) {
-
-            require_once(\TYPO3\CMS\Core\Utility\GeneralUtility::getFileAbsFileName('EXT:'.self::$extKey.'/lib/SolrPhpClient/Apache/Solr/Service.php'));
-
-        }
-
-        $solrInfo = self::getSolrConnectionInfo($core);
+        $solrInfo = self::getSolrConnectionInfo();
+
+        $config = array (
+            'endpoint' => array (
+                'dlf' => array (
+                    'scheme' => $solrInfo['scheme'],
+                    'host' => $solrInfo['host'],
+                    'port' => $solrInfo['port'],
+                    'path' => '/'.$solrInfo['path'].'/',
+                    'core' => $core,
+                    'username' => $solrInfo['username'],
+                    'password' => $solrInfo['password']
+                )
```

Comment:
We need here another setting "timeout". The default "timeout" is set to 5 seconds by Solarium (see `Core/Client/Endpoint.php`). Most METS files referencing FULLTEXTS cannot be parsed in my setup within 5 seconds. 
Error message is: "Solr HTTP error: HTTP request failed, Operation timed out after 5001 milliseconds with 0 bytes "received
I played around and found 10s ok. But this should be configurable, I think.
It's quite easy, to pass the timeout parameter to the Solarium:

<pre>
$config = array (
            'endpoint' => array (
                'dlf' => array (
                    'scheme' => $solrInfo['scheme'],
                    'host' => $solrInfo['host'],
                    'port' => $solrInfo['port'],
                    'path' => '/'.$solrInfo['path'].'/',
                    'core' => $core,
                    'username' => $solrInfo['username'],
                    'password' => $solrInfo['password'],
                     'timeout' => 10
                )
            )
        );

</pre>

My example: http://digital.slub-dresden.de/data/goobi/490223001-1916092402/DresneNa_490223001-1916092402_02-a_mets.xml